swift/send-message/message-types/email/Sources/email_message.swift

```
@@ -0,0 +1,48 @@
+import AsyncHTTPClient
+import Foundation
+import NIO
+
+class EmailMessenger: Messenger{
+
+    private let mailgunAPIKey: String
+    private let mailgunDomain: String
+    private let httpClient: HTTPClient
+
+    init(_ env_vars: [String: String]) throws{
+        guard let mailgunAPIKey = env_vars["MAILGUN_API_KEY"],
+            let mailgunDomain = env_vars["MAILGUN_DOMAIN"] else {
+            throw MessengerError.misconfigurationError(error: "Missing environment variables.")
+        }
+        self.mailgunAPIKey = mailgunAPIKey
+        self.mailgunDomain = mailgunDomain
+        httpClient = HTTPClient(eventLoopGroupProvider: .createNew)
+    }
+    
+    public func sendMessage(messageRequest: Message) async -> Error? {     
+        let targetURL:String = "https://api.mailgun.net/v3/\(mailgunDomain)/messages"
+        let auth:String = "api:\(mailgunAPIKey)".data(using: .utf8)!.base64EncodedString()
+        var request = HTTPClientRequest(url: targetURL)
+        request.method = .POST
+        request.headers.add(name: "Content-Type", value: "application/x-www-form-urlencoded")
+        request.headers.add(name: "Authorization", value: "Basic \(auth)")
+
+        let bodyString: String = "from=me@samples.mailgun.org&to=\(messageRequest.recipient)&subject=\(messageRequest.subject ?? "")&text=\(messageRequest.content)"
+        request.body = .bytes(ByteBuffer(bytes: Data(bodyString.utf8)))
+        do{
+            let response = try await httpClient.execute(request, timeout: .seconds(30))
+            if response.status == .ok {       
+                return nil
+            } else {        
+                return MessengerError.validationError(error: "Unable to send email, API Status Code: \(response.status)")
```

make sure we're not swallowing mailgun errors